src/index.js

```diff
@@ -0,0 +1,204 @@
+          clean: {
+            type: 'entrypoint',
+            lifecycleEvents: [
+              'clean',
+            ],
+          },
+        },
+      },
+    };
+
+    /* istanbul ignore next */
+    this.hooks = {
+      'after:deploy:deploy': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:deploy'))
+        .then(() => this.sls.pluginManager.spawn('ignitor:clean')),
+
+      'before:package:createDeploymentArtifacts': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:schedule'))
+        .then(() => this.sls.pluginManager.spawn('ignitor:wrap')),
+
+      'after:package:createDeploymentArtifacts': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:clean')),
+
+      'before:deploy:function:packageFunction': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:schedule'))
+        .then(() => this.sls.pluginManager.spawn('ignitor:wrap')),
+
+      'before:invoke:local:invoke': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:wrap')),
+
+      'after:invoke:local:invoke': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:clean')),
+
+      'before:run:run': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:wrap')),
+
+      'after:run:run': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:clean')),
+
+      // used when debugging ignitor via command serverless ignitor
+      'ignitor:ignitor': () => bPromise.bind(this)
+        .then(build.prebuild)
+        .then(this.schedule)
+        .then(this.wrap),
+
+      'ignitor:schedule:schedule': () => bPromise.bind(this)
+        .then(this.schedule),
+
+      'ignitor:wrap:wrap': () => bPromise.bind(this)
+        .then(build.prebuild)
+        .then(this.wrap),
+
+      'ignitor:deploy:deploy': () => bPromise.bind(this)
+        .then(this.deploy),
+
+      'ignitor:clean:clean': () => bPromise.bind(this)
+        .then(build.clean),
+    };
+  }
+
+  schedule() {
+    const options = get(this, 'sls.service.custom.ignitor', []);
+    const keys = options.length === 0
+      ? new RegExp('.*', 'g')
+      : new RegExp(
+        options.map((k) => k.replace(/\//g, '')).join('|'),
+        'g',
+      );
+
+    this.service = get(this, 'sls.service.service.name', get(this, 'sls.service.service'));
+    this.stage = get(this, 'optStage', get(this, 'sls.service.provider.stage', '*'));
+    this.region = get(this, 'optRegion', get(this, 'sls.service.provider.region'));
+    this.scheduled = Object.keys(this.sls.service.functions)
+      .filter((name) => name.match(keys));
+
+    if (!this.sls.service.resources) {
+      this.sls.service.resources = { Resources: {} };
+    }
+
+    role.attachRoleToLambda(this.sls.service.functions.ignitorDelegate);
+    role.createLambdaRole(this.sls.service.resources.Resources, {
```

Comment:
This is kind of ugly, would like to clean this up, but can probably be done at a later date.